rust_src/src/frames.rs

```diff
@@ -661,6 +783,25 @@ pub fn visible_frame_list() -> LispObject {
     filter_frame_list(LispFrameRef::is_visible)
 }
 
+/// Select FRAME.
+/// Subsequent editing commands apply to its selected window.
+/// Optional argument NORECORD means to neither change the order of
+/// recently selected windows nor the buffer list.
+///
+/// The selection of FRAME lasts until the next time the user does
+/// something to select a different frame, or until the next time
+/// this function is called.  If you are using a window system, the
+/// previously selected frame may be restored as the selected frame
+/// when returning to the command loop, because it still may have
+/// the window system's input focus.  On a text terminal, the next
+/// redisplay will display FRAME.
+///
+/// This function returns FRAME, or nil if FRAME has been deleted.
+#[lisp_fn(min = "1", intspec = "e")]
+pub fn select_frame(frame: LispObject, norecord: LispObject) -> LispObject {
+    do_switch_frame(frame, 1, 0, norecord)
```

Comment:
This can become `do_switch_frame_rust(frame, true, false, norecord).into()` once you make the Rust version.